src/drivers/imu/bmi088/BMI088_accel.cpp

```diff
@@ -0,0 +1,597 @@
+    uint8_t cmd[4] = { (uint8_t)(reg | DIR_READ), 0, 0, 0 };
+
+    transfer(cmd, cmd, sizeof(cmd));
+
+    return (uint16_t)(cmd[2] << 8) | cmd[3]; // Skip dummy byte in cmd[1]
+}
+
+int BMI088_accel::reset()
+{
+    write_reg(BMI088_ACC_SOFTRESET, BMI088_SOFT_RESET); // Soft-reset
+    up_udelay(5000); // After a POR or soft-reset, the sensor needs up to 1ms boot time (see section "Power Modes: Accelerometer" in the BMI088 datasheet). Based off of testing it seems this value needs to be increased from 1ms. Setting it to 5ms.
+
+    // Perform a dummy read here to put the accelerometer part of the BMI088 back into SPI mode after the reset
+    // The dummy read basically pulls the chip select line low and then high
+    // See section "Serial Peripheral Interface (SPI)" of the BMI088 datasheet for more details.
+    read_reg(BMI088_ACC_CHIP_ID);
+
+    // Enable Accelerometer
+    // The accelerometer needs to be enabled first, before writing to its registers
+    write_checked_reg(BMI088_ACC_PWR_CTRL, BMI088_ACC_PWR_CTRL_EN);
+    up_udelay(5000); // After changing power modes, the sensor requires up to 5ms to settle. Any communication with the sensor during this time should be avoided (see section "Power Modes: Acceleromter" in the BMI datasheet)
+
+    // Set the PWR CONF to be active
+    write_checked_reg(BMI088_ACC_PWR_CONF, BMI088_ACC_PWR_CONF_ACTIVE); // Sets the accelerometer to active mode
+
+    // Write accel bandwidth and output data rate
+    // ToDo set the bandwidth
+    accel_set_sample_rate(BMI088_ACCEL_DEFAULT_RATE); //set accel ODR
+    set_accel_range(BMI088_ACCEL_DEFAULT_RANGE_G); //set accel range
+
+    // Configure the accel INT1
+    write_checked_reg(BMI088_ACC_INT1_IO_CONF, BMI088_ACC_INT1_IO_CONF_INT1_OUT | BMI088_ACC_INT1_IO_CONF_PP | BMI088_ACC_INT1_IO_CONF_ACTIVE_HIGH); // Configure INT1 pin as output, push-pull, active high
+    write_checked_reg(BMI088_ACC_INT1_INT2_MAP_DATA, BMI088_ACC_INT1_INT2_MAP_DATA_INT1_DRDY); // Map DRDY interrupt on pin INT1
+
+    uint8_t retries = 10;
+
+    while (retries--) {
+        bool all_ok = true;
+
+        for (uint8_t i = 0; i < BMI088_ACCEL_NUM_CHECKED_REGISTERS; i++) {
+            if (read_reg(_checked_registers[i]) != _checked_values[i]) {
+                write_reg(_checked_registers[i], _checked_values[i]);
+                all_ok = false;
+            }
+        }
+
+        if (all_ok) {
+            break;
+        }
+    }
+
+    return OK;
+}
+
+int
+BMI088_accel::probe()
+{
+    // Perform a dummy read here to put the accelerometer part of the BMI088 back into SPI mode after the reset
+    // The dummy read basically pulls the chip select line low and then high
+    // See section "Serial Peripheral Interface (SPI)" of the BMI088 datasheet for more details.
+    read_reg(BMI088_ACC_CHIP_ID);
+
```

Comment:
According to the BMI088 datasheet I need to transition the CS pin from low to high of the accelerometer in order to switch it to SPI mode (It starts in I2C mode). Since I don't have access to the _dev pointer of the SPI device (it is declared as private in SPI.h), I can't call SPI_SELECT to just set the pin high and low. Therefore I do a dummy read operation which has the side effect of pulling the CS pin low and then high again after the read operation has completed. (See section "3. Quick Start Guide - Device Initialization").